gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -2928,3 +2928,45 @@ nnstreamer_filter_shared_model_remove (void *instance, const char *key,
 
   return ret;
 }
+
+/* extern functions for shared model representation */
+/**
+ * @brief Helper to reload interpreter for instances that has shared key.
+ *        `replace_callback` is called iterating instances in referred list.
+ * @param[in] instance The instance that is sharing the model representation.
+ * @param[in] key The key to find the shared model.
+ * @param[in] interpreter The new interpreter to replace.
+ * @param[in] replace_callback The callback function to replace with new interpreter.
+ * @param[in] free_callback The callback function to destroy the old interpreter.
+ */
+void
+nnstreamer_filter_shared_model_replace (void *instance, const char *key,
+    void *new_interpreter, void (*replace_callback) (void *, void *),
+    void (*free_callback) (void *))
+{
+  GstTensorFilterSharedModelRepresenatation *model_rep;
+  GList *itr;
+  UNUSED (instance);
+
+  if (!shared_model_table) {
+    ml_loge ("The shared model representation is not supported properly!");
+    return;
+  }
+  if (!key) {
+    ml_loge ("The key should NOT be NULL!");
+    return;
+  }
+
+  g_mutex_lock (&mutex);
```

Comment:
as `replace_callback` is only used here, it seems it's protected and looks valid.